Create GitHub Actions Workflow for Automated Git Rebase Operations with YAML Configuration (#2)
anatawa12/copilot-workspace: .github/instructions/automated-rebase.instructions.md

---
applyTo: '**/*'
---

# GitHub Copilot Instructions for Automated Rebase

This repository supports automated git rebase operations through GitHub Actions. You can help users create and manage rebase operations by generating the appropriate `.github/rebase-config.yml` files using the YAML format.

## When to Use Automated Rebase

Users might want to use automated rebase when they need to:
- Squash multiple commits into one with a cleaner commit message
- Combine feature commits before merging to main branch
- Clean up commit history by combining related changes
- Prepare commits for a pull request review
- Drop initial plan commits or work-in-progress commits that are no longer needed

> **Important Notes:**
> - For normal operations, split your changes into separate commits: one for your actual changes, and one for the `rebase-config.yml` file
> - The `rebase-config.yml` commit should be the last commit since any commits after it will be removed during the rebase process

Reviewer: ```suggestion
**Important Notes:**
- The `rebase-config.yml` commit should be the last commit without any other change since any commits after it will be removed during the rebase process
```
